openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/documents/DocumentCategoriesWebScript.java

```diff
@@ -0,0 +1,86 @@
+import dk.openesdh.repo.services.documents.DocumentCategoryService;
+import dk.openesdh.repo.webscripts.AbstractRESTWebscript;
+import dk.openesdh.repo.webscripts.ParamUtils;
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.codehaus.plexus.util.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.springframework.extensions.webscripts.WebScriptException;
+import org.springframework.extensions.webscripts.WebScriptRequest;
+import org.springframework.extensions.webscripts.WebScriptResponse;
+
+public class DocumentCategoriesWebScript extends AbstractRESTWebscript {
+
+    private DocumentCategoryService documentCategoryService;
+
+    public void setDocumentCategoryService(DocumentCategoryService documentCategoryService) {
+        this.documentCategoryService = documentCategoryService;
+    }
+
+    @Override
+    protected NodeRef getNodeRef(WebScriptRequest req, Map<String, String> templateArgs) {
+        String nodeRefId = req.getParameter("nodeRefId");
+        if (StringUtils.isNotEmpty(nodeRefId)) {
+            return new NodeRef(nodeRefId);
+        }
+        return super.getNodeRef(req, templateArgs);
+    }
+
+
+    @Override
+    protected void get(NodeRef nodeRef, WebScriptRequest req, WebScriptResponse res) throws IOException, JSONException {
+        if (nodeRef == null) {
+            new JSONArray(documentCategoryService.getDocumentCategories().stream()
+                    .map(DocumentCategory::toJSONObject)
+                    .collect(Collectors.toList())
+            ).write(res.getWriter());
+            return;
+        }
+        DocumentCategory documentCategory = documentCategoryService.getDocumentCategory(nodeRef);
+        if (documentCategory == null) {
+            throw new WebScriptException("Document category not found");
+        }
+        writeDocumentCategoryToResponse(documentCategory, res);
+    }
+
+    @Override
+    protected void post(NodeRef nodeRef, WebScriptRequest req, WebScriptResponse res) throws IOException, JSONException {
+        String name = ParamUtils.getRequiredParameter(req, "name");
+        String displayName = ParamUtils.getRequiredParameter(req, "displayName");
+        DocumentCategory saveDocumentCategory = updateDocumentCategory(nodeRef, name, displayName);
+        writeDocumentCategoryToResponse(saveDocumentCategory, res);
+    }
+
+    @Override
+    protected void delete(NodeRef nodeRef, WebScriptRequest req, WebScriptResponse res) throws IOException, JSONException {
+        checkRequiredNodeRef(nodeRef);
+        DocumentCategory category = new DocumentCategory();
+        category.setNodeRef(nodeRef);
+        documentCategoryService.deleteDocumentCategory(category);
+        res.getWriter().append("Deleted succesfully");
+    }
+
+    private void checkRequiredNodeRef(NodeRef nodeRef) throws WebScriptException {
```

Comment:
The method should be extracted to the super class to avoid code duplication. The DocumentTypeWebScript contains the same method.